### DISCUSSION TITLE
making a pipeline with PyODAnnotator

### DISCUSSION THREAD

**Discussion by hstarmans** [alan-turing-institute/sktime#6576] [Q&A]
Challenge, I can't make pipes with PyODAnnotator
```python
from sklearn.preprocessing import StandardScaler
from sktime.annotation.adapters import PyODAnnotator
from pyod.models.knn import KNN
StandardScaler()*PyODAnnotator(KNN(0.05))
```
Results in
```
TypeError: unsupported operand type(s) for *: 'StandardScaler' and 'PyODAnnotator'"
```
Current solution is graph pipeline with a linear structure.
The Id transformer converts StandardScaler from a sklearn object into a skobject with a dunder method.
```python
from sktime.transformations.compose import Id
from sktime.pipeline import Pipeline
general_pipeline = Pipeline()
for step in [
    {"skobject": Id()*StandardScaler(), "name": "sdsc", "edges": {"X": "X", 'y':'y'}},
    {"skobject": PyODAnnotator(KNN(0.05)), "edges": {"X": "sdsc", 'y':'sdsc'},
     "name": "knn",}]:
    general_pipeline = general_pipeline.add_step(**step)
```
Although, I get a pipe it doesn't work, so guess this also doesn't count as solution.
Is there a better solution?

**Comment by fkiraly** [comment]
We are currently reworking the time series annotation module, dunders have not been implemented for these yet.

Since 0.30.0, the annotators provide a `transform` method, so using `sklearn` (not `sktime`) `make_pipeline` might work, and I thought the `Pipeline` should also work?

FYI @benHeid re pipeline, FYI @Alex-JG3 re annotation estimators.

**Comment by fkiraly** [comment]
I've opened an issue on how pipelines should behave:
https://github.com/sktime/sktime/issues/6591

What would be much appreciated there is your input, @hstarmans, on how the pipeline *should* behave. Is what I outline correct?

Or would you rather expect the output of `predict`?
